src/main/java/com/teambash/keepinmind/service/impl/SynonymServiceImp.java

```diff
@@ -0,0 +1,48 @@
+package com.teambash.keepinmind.service.impl;
+
+import com.teambash.keepinmind.model.Synonym;
+import com.teambash.keepinmind.repository.SynonymRepository;
+import com.teambash.keepinmind.service.SynonymService;
+import com.teambash.keepinmind.shared.dto.SynonymDto;
+import lombok.extern.java.Log;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Log
+public class SynonymServiceImp implements SynonymService {
+
+  @Autowired private SynonymRepository synonymRepository;
+
+  @Autowired private ModelMapper modelMapper;
```

Comment:
```suggestion
  private final SynonymRepository synonymRepository;
  private final ModelMapper modelMapper;
  
  public SynonymServiceImp(SynonymRepository synonymRepository, ModelMapper modelMapper) {
    this.synonymRepository = synonymRepository;
    this.modelMapper = new ModelMapper();
  }
```

This can be changed this way. Since we only have one constructor, we don't need to specify the Autowired annotation, the injection is implicit. It allows us to make the props final as well.